lib/cog/models/group.ex

```diff
@@ -28,12 +29,30 @@ defmodule Cog.Models.Group do
   detail_fields [:id, :name]
 
   @doc """
-  Creates a changeset based on the `model` and `params`.
-
   If no params are provided, an invalid changeset is returned
   with no validation performed.
   """
-  def changeset(model, params \\ :empty) do
+  def changeset(model), do: changeset(model, :empty)
+
+  @doc """
+  Creates a changeset based on the `model` to validate a delete
+  action.
+  """
+  def changeset(model, :delete) do
+    changeset = %{Changeset.change(model) | action: :delete}
+
+    if model.name == Cog.admin_group do
+      changeset
+      |> add_error(:name, "admin group may not be modified")
```

Comment:
We could have changeset functions that bail if you try to delete or rename a protected "thing", which would provide protection in Elixir, since recovering from trigger-thrown exceptions in a nice customizable way with Ecto / Postgrex is not as straightforward as I'd like (at least, it wasn't back when I tried to do it a few months ago; not sure if anything has changed on that front since then).

Triggers could provide the last line of defence, should we ever hit the database without going through the app.